src/encoded/audit/experiment.py

```diff
@@ -2534,11 +2547,23 @@ def audit_experiment_ChIP_control(value, system, files_structure):
             # The binding group agreed that ChIP-seqs all should have an input control.
             detail = 'Experiment {} is ChIP-seq and requires at least one input control, as agreed upon by the binding group. {} is not an input control'.format(
                 value['@id'],
-                control['@id'])
+                control_dataset['@id'])
             yield AuditFailure('missing input control', detail, level='NOT_COMPLIANT')
     return
 
 
+def is_control_dataset(dataset):
```

Comment:
One suggestion is to add some unit tests for your added functions `is_matching_biosample_control` and `is_control_dataset`. That way I would get a better understanding of what these are supposed to do and there would be some concrete use cases that you know work as expected.